clightning/clightning.go

```diff
@@ -619,6 +590,63 @@ func (cl *ClightningClient) GetPeers() []string {
 	return peerlist
 }
 
+// ProbePayment trying to pay via a route with a random payment hash
+// that the receiver doesn't have the preimage of.
+// The receiver node aren't able to settle the payment.
+// When the probe is successful, the receiver will return
+// a incorrect_or_unknown_payment_details error to the sender.
+func (cl *ClightningClient) ProbePayment(scid string, amountMsat uint64) (bool, string, error) {
+	var res ListPeerChannelsResponse
+	err := cl.glightning.Request(ListPeerChannelsRequest{}, &res)
+	if err != nil {
+		return false, "", fmt.Errorf("ListPeerChannelsRequest() %w", err)
+	}
+	var channel PeerChannel
+	for _, ch := range res.Channels {
+		if ch.ShortChannelId == lightning.Scid(scid).ClnStyle() {
+			if err := cl.checkChannel(ch); err != nil {
+				return false, "", err
+			}
+			channel = ch
+		}
+	}
+
+	route, err := cl.glightning.GetRoute(channel.PeerId, amountMsat, 1, 0, cl.nodeId, 0, nil, 1)
```

Comment:
Yes, I thought it would be better.

Like `PayInvoiceViaChannel`, I have modified it to ensure that invoice payments are probed through a direct channel to peers.